kineticTypo/kineticTypo1/Tile.pde

```diff
@@ -1,102 +1,100 @@
 class Tile {
-
- float tx;
- float ty;
   
- int tilesX; 
- int tilesY;
+  // Since we configure this at the beginning we can make this
+  // variable a final static variable, this means we know 
+  // that this variable will never change
+  private final static int fraqX = 4; 
+  private final static int fraqY = 6;
+  private final static int tolleranceX = 30;
  
- float tileW; 
- float tileH;
-
- float fraqX; 
- float fraqY;
+  // declaring this variables as private makes sure we 
+  // can only access this variables within the Tile class
+  private int offsetX;
  
- float fraqW;
- float fraqH; 
+  private int tileW; 
+  private int tileH;
  
- Tile (){ 
-   
-   tx = 0;
-   ty = 0;
+  private int fraqW;
+  private int fraqH; 
+  
+  private int posX;
+  private int posY;
+  
+  private float mx = 0;
+  private float my = 0;
+  
+  private PImage originalImage; 
+  private PGraphics pg;
  
-   tilesX = 2; // how many tiles on the Y/X axis
-   tilesY = 1;
+  Tile (PImage img, int x, int widthTile, int heightTile) {
+    this.originalImage = img;
+    
+    this.offsetX = x;
    
-   tileW = width/tilesX;
-   tileH = height/tilesY;
- 
- }
- 
- Tile (int amountTX, int amountTY){ // constructor
- 
-   tx = 0;
-   ty = 0;
- 
-   tilesX = amountTX; // how many tiles on the Y/X axis
-   tilesY = amountTY;
+    this.tileW = widthTile;
+    this.tileH = heightTile;
    
-   tileW = width/tilesX;
-   tileH = height/tilesY;
- 
- }
- 
- void display(){
- 
-  //fill(255);
-  //noStroke();
-  //rect(tx,ty, tileW, tileH);
-  //test();
-  fragment();
- }
- 
- void fragment(){
- 
-  fraqX = 4; // how many tiles on the Y/X axis  
-  fraqY = 4; 
+    this.posX = this.offsetX * this.tileW;
+    this.posY = 0;
+   
+    this.pg = createGraphics(this.tileW, this.tileH);
+   
+    // calcultaion of fraqments
+    this.fraqW = int(this.tileW / Tile.fraqX);  
+    this.fraqH = int(this.tileH / Tile.fraqY);
```

Comment:
Those variables can be calculated once in the constructor. We don't need to change the on each draw. 

> Note that we access `fraqX` and `fraqY` via the `Tile` class and not `this`. This is because those variables are `static` properties of the Class, because it is the same value for each Tile.